scripts/zenodo-bibtex.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python
+"""This script forms part of the internal workings of Firedrake-Zenodo. It is not intended for direct user interaction."""
+import cgi
+import cgitb
+cgitb.enable()
+import urllib2
+import json
+from HTMLParser import HTMLParser
+import sys
+
+
+def fail(code, reason):
+
+    sys.stdout.write('Status: %s\r\n\r\n%s\r\n\r\n' % (code, reason))
+    sys.exit(1)
+
+def find_ids(response, firedrake_tag):
+
+    ids = []
+    for record in response:
+        try:
+            for id in record['metadata']['related_identifiers']:
+                if id['identifier'].endswith(firedrake_tag):
+                    ids.append(record['record_id'])
+        except KeyError:
```

Comment:
So you spin over the json blobs provided by zenodo listing your depositions and collect those that match the provided tag.